### PR TITLE
[Fix] Pass diffmode in execution

### PR DESCRIPTION
**Pull request by dominikandreasseitz**

### CODE REVIEW

**Review by awennersteen — APPROVED** [review]
lgtm

File: qadence/backends/pytorch_wrapper.py [hunk 1]
```diff
@@ -315,13 +315,16 @@ def convert(
         observable: list[AbstractBlock] | AbstractBlock | None = None,
     ) -> Converted:
         if self.diff_mode != DiffMode.AD and observable is not None:
+            msg = (
+                f"Differentiation mode '{self.diff_mode}' does not support parametric observables."
```
Comment (awennersteen) [inline qadence/backends/pytorch_wrapper.py:319]:
```suggestion
                f"Differentiation mode {self.diff_mode} does not support parametric observables."
```

Comment (awennersteen) [reply qadence/backends/pytorch_wrapper.py:319]: with f strings that works, right?

Comment (dominikandreasseitz) [reply qadence/backends/pytorch_wrapper.py:319]: i prefer "ValueError: Differentiation mode 'adjoint' does not support parametric observables." (so the quote, if you dont mind)

**Pull request merged**